source/write/insert.txt

```diff
@@ -168,6 +211,140 @@ document-level validation.
 
    sample_restaurants.restaurants.insert_many(document_list, bypass_document_validation = True)
 
+Troubleshooting
+---------------
+
+.. include:: /includes/type-hints/troubleshooting-client-type.rst
+
+.. include:: /includes/type-hints/troubleshooting-incompatible-type.rst
+
+You might see a similar error if you pass a list to the ``insert_one()`` method:
+
+.. code-block:: bash
+
+   error: Argument 1 to "insert_one" of "Collection" has
+   incompatible type "List[Dict[<nothing>, <nothing>]]";
+   expected "Mapping[str, Any]"
+
+This error occurs because the ``insert_one()`` method accepts a document, not a list.
+You can resolve this error by passing a document to the ``insert_one()`` method or by
+calling the ``insert_many()`` method instead.
+
+TypedDict Missing _id Key
```

Comment:
Q for tech review: Is one of these three solutions recommended over the others?

Reply:
@mongoKart I love the fact that you've laid out 3 solutions and the tradeoffs. Different programmers will have their own aesthetics and constraints. It *does* make me wonder again why it was chosen to distribute, not consolidate, the type hinting discussion?